test/utils.py

```diff
@@ -507,6 +507,49 @@ def compare_matmat(
         assert allclose_report(o_x, m_x, **tol)
 
 
+def compare_consecutive_matmats(
+    op: PyTorchLinearOperator,
+    adjoint: bool,
+    is_vec: bool,
+    num_vecs: int = 2,
+    rtol: float = 1e-5,
+    atol: float = 1e-8,
+):
+    """Compare applying the linear operator to two identical vectors in sequence.
+
+    Args:
+        op: The operator to test.
+        adjoint: Whether to test the adjoint operator.
+        is_vec: Whether to test matrix-vector or matrix-matrix multiplication.
+        num_vecs: Number of vectors to test (ignored if ``is_vec`` is ``True``).
+            Default: ``2``.
+        rtol: Relative tolerance for the comparison. Default: ``1e-5``.
+        atol: Absolute tolerance for the comparison. Default: ``1e-8``.
+    """
+    if adjoint:
```

Comment:
Is there a reason to offer this `adjoint` option? I think it might be more explicit to just require the user to pass `op.adjoint()` to this function.

For example, when I first saw a call to this function with `adjoint=True`, I wasn't sure if it will compare the matmat for the linop AND its adjoint or only for the latter (which turns out to be the case).